lnwallet/btcwallet/psbt.go

```diff
@@ -352,6 +369,72 @@ func validateSigningMethod(in *psbt.PInput) (input.SignMethod, error) {
 	}
 }
 
+// EstimateInputWeight estimates the weight of a PSBT input and adds it to the
+// passed in TxWeightEstimator. It returns an error if the input type is
+// unknown or unsupported. Only inputs that have a known witness size are
+// supported, which is P2WKH, NP2WKH and P2TR (key spend path).
+func EstimateInputWeight(in *psbt.PInput, w *input.TxWeightEstimator) error {
+	if in.WitnessUtxo == nil {
+		return ErrInputMissingUTXOInfo
+	}
+
+	pkScript := in.WitnessUtxo.PkScript
+	switch {
+	case txscript.IsPayToScriptHash(pkScript):
+		w.AddNestedP2WKHInput()
```

Comment:
Yes, theoretically. But because the `lnd` wallet only supports NP2WKH and no bare/standard P2SH it will never select or be able to sign for such an input. And we wouldn't know the length of the witness anyway, so we couldn't make a better guess. But I think we can just assume nobody is going to use a P2SH input foreign to `lnd` in this RPC? And if they do, the fee estimation will just be off...